src/core/raw.c

```diff
@@ -174,6 +174,15 @@ raw_input(struct pbuf *p, struct netif *inp)
         void *old_payload = p->payload;
 #endif
         ret = RAW_INPUT_DELIVERED;
+
+#if LWIP_IPV6
+        u8_t iphdr_version;
+        iphdr_version = IP_HDR_GET_VERSION(p->payload);
+        if (iphdr_version == 6) {
+          pbuf_remove_header(p, IP6_HLEN);
```

Comment:
I'm not sure this is fully-valid as you're skipping only the ipv6 main header, not all of the optional headers (see big while / switch before entering this function: https://github.com/phoenix-rtos/lwip/blob/STABLE-2_1_x_phoenix/src/core/ipv6/ip6.c#L1053). The RFC says that we should remove also all extension headers 
 (*the application will only receive the payload i.e., the data after the  IPv6 header and all the extension headers*)

Also, add some comments - eg. that it's a temporary workaround (we're stripping headers at a wrong place - before we know whether the user user `recv` or `recvmsg` interface - so this change makes it impossible to implement `recvmsg` correctly). Or try to do it in a correct place (proabbly tricker  :/).

Please update also the commit description, something along the lines:

RFCxxxx says that `recvfrom` should return only IPv6 packet payload without IPv6 and extension headers. 

To receive IPv6 headers `recvmsg` interface should be used (not implemented).

Reply:
Fixed this. Now we're skipping all the extension headers as well.